torch/onnx/_internal/fx/passes/modularization.py

```diff
@@ -0,0 +1,648 @@
+    def from_fx_tracer_produced_raw_meta(
+        cls, raw_meta: Tuple[str, Type]
+    ) -> _ModuleMeta:
+        """Create a module meta from raw meta produced by FX symbolic tracer."""
+        module_name, module_class = raw_meta
+        return _ModuleMeta(module_name, module_class, raw_meta)
+
+    @classmethod
+    def from_dynamo_produced_raw_meta(
+        cls, raw_meta: Tuple[str, Tuple[str, Type]]
+    ) -> _ModuleMeta:
+        """Create a module meta from raw meta produced by FX dynamo tracer."""
+        module_name, (qualified_name, module_class) = raw_meta
+        return _ModuleMeta(module_name, module_class, raw_meta)
+
+
+class _ModuleStackMeta:
+    """Meta information about the module stack.
+
+    This class is used to represent the module stack information in a more
+    structured way. It parses raw module stack information from node.meta.
+    """
+
+    _module_stack: Final[List[_ModuleMeta]]
+
+    @_beartype.beartype
+    def __init__(
+        self,
+        nn_module_stack_meta: Optional[
+            Union[
+                _FX_TRACER_NN_MODULE_STACK_META_TYPE, _DYNAMO_NN_MODULE_STACK_META_TYPE
+            ]
+        ],
+    ):
+        self._module_stack = []
+        if nn_module_stack_meta is None:
+            return
+        raw_meta = copy.copy(nn_module_stack_meta)
+        is_fx_tracer_produced_raw_meta = isinstance(
+            raw_meta, _FX_TRACER_NN_MODULE_STACK_META_TYPE
+        )
+        is_dynamo_produced_raw_meta = (
+            isinstance(raw_meta, dict) and not is_fx_tracer_produced_raw_meta
+        )
+        for item in raw_meta.items():
+            if is_fx_tracer_produced_raw_meta:
+                self.push(_ModuleMeta.from_fx_tracer_produced_raw_meta(item))
+            elif is_dynamo_produced_raw_meta:
+                self.push(_ModuleMeta.from_dynamo_produced_raw_meta(item))
+            else:
+                raise AssertionError(
+                    f"Unknown type of nn_module_stack_meta: {type(raw_meta)}"
+                )
+
+    def __len__(self) -> int:
+        return len(self._module_stack)
+
+    def __getitem__(self, index: int) -> _ModuleMeta:
+        return self._module_stack[index]
+
+    def __iter__(self) -> Iterator[_ModuleMeta]:
+        return iter(self._module_stack)
+
+    def is_empty_or_root(self) -> bool:
+        return len(self._module_stack) == 0
+
+    def top(self) -> _ModuleMeta:
+        """Returns the top module meta in the stack. I.e., the meta for leaf module."""
+        if self.is_empty_or_root():
+            return _ModuleMeta.root()
+        return self._module_stack[-1]
+
+    @_beartype.beartype
+    def is_superset_of(
+        self,
+        module_stack: _ModuleStackMeta,
+    ) -> bool:
+        """Determines if self is a superset of the provided module stack.
+
+        I.e., If self includes all elements from the provided module stack, plus additional
+        elements op top. If self is empty or root, this method always return False.
```

Comment:
```suggestion
        elements on top. If self is empty or root, this method always return False.
```
It's better to give one example here. There is no official definition of `module stack`.

Reply:
done